### ISSUE TITLE
Indendation rule bugged

### ISSUE THREAD

**Issue by adamreisnz** [eslint/eslint#9020]
**Tell us about your environment**

* **ESLint Version:** 4.3.0
* **Node Version:** 8.2.1

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
root: true
extends: eslint:recommended
parserOptions:
  ecmaVersion: 2017
  ecmaFeatures:
    impliedStrict: true
  sourceType: module
env:
  es6: true
  browser: true
  node: true
  mocha: true
  jasmine: true
globals:
  angular: false
  Raven: false
rules:
  linebreak-style:
    - error
    - unix
  max-len:
    - warn
    - code: 80
      tabWidth: 2
      ignoreComments: true
      ignoreUrls: true
      ignoreTemplateLiterals: true
      ignoreRegExpLiterals: true
  indent:
    - error
    - 2
    - SwitchCase: 1
      VariableDeclarator:
        let: 2
        const: 3
  semi:
    - error
    - always
  consistent-this:
    - error
    - self
    - $ctrl
  quotes:
    - error
    - single
    - allowTemplateLiterals: true
  quote-props:
    - error
    - as-needed
  curly:
    - error
    - all
  comma-dangle:
    - error
    - always-multiline
  new-cap:
    - error
    - newIsCap: true
      capIsNew: true
      properties: false
  camelcase:
    - error
    - properties: never
  array-bracket-spacing:
    - error
    - never
  arrow-spacing:
    - error
    - before: true
      after: true
  block-spacing:
    - error
    - always
  comma-spacing:
    - error
    - before: false
      after: true
  computed-property-spacing:
    - error
    - never
  generator-star-spacing:
    - error
    - before: true
      after: false
  key-spacing:
    - error
    - beforeColon: false
      afterColon: true
      mode: minimum
  keyword-spacing:
    - error
    - before: true
  semi-spacing:
    - error
    - before: false
      after: true
  space-in-parens:
    - error
    - never
  space-unary-ops:
    - error
    - words: true
      nonwords: false
  space-before-function-paren:
    - error
    - never
  space-before-blocks:
    - error
    - always
  yoda:
    - error
    - never
  wrap-iife:
    - error
    - outside
  eqeqeq:
    - error
    - always
  newline-per-chained-call:
    - error
    - ignoreChainWithDepth: 3
  one-var-declaration-per-line:
    - error
    - initializations
  brace-style:
    - error
    - stroustrup
  no-implicit-coercion:
    - error
    - boolean: false
  no-multiple-empty-lines:
    - error
    - max: 1
  no-unused-vars:
    - error
    - vars: all
      args: all
  eol-last: error
  dot-notation: error
  space-infix-ops: error
  no-with: error
  no-unreachable: error
  no-redeclare: error
  no-unexpected-multiline: error
  no-multi-spaces: error
  no-multi-str: error
  no-trailing-spaces: error
  no-mixed-spaces-and-tabs: error
  no-spaced-func: error
  no-whitespace-before-property: error
  no-lonely-if: error
  no-var: error
  no-console: warn

```

</details>

**What did you do? Please include the actual source code causing the issue.**
I updated ESLint from 3.x to 4.x

```js

/**
 * Module definition and dependencies
 */
angular.module('App.Portal.Welcome.Credentials', [])

/**
 * Config
 */
.config($stateProvider => {
  $stateProvider.state('welcome.credentials', {
    url: '/credentials',
    component: 'welcomeCredentialsRoute',
  });
})

/**
 * Route component
 */
.component('welcomeCredentialsRoute', {
  controller: 'WelcomeCredentialsCtrl',
  templateUrl: 'portal/welcome/steps/credentials.html',
  bindings: {
    token: '<',
    club: '<',
    user: '<',
  },
})
```

**What did you expect to happen?**
For ESLint to leave my code untouched, as it's properly indented.

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint indents it as follows:

```js
/**
 * Module definition and dependencies
 */
angular.module('App.Portal.Welcome.Credentials', [])

/**
 * Config
 */
  .config($stateProvider => {
    $stateProvider.state('welcome.credentials', {
      url: '/credentials',
      component: 'welcomeCredentialsRoute',
    });
  })

/**
 * Route component
 */
  .component('welcomeCredentialsRoute', {
    controller: 'WelcomeCredentialsCtrl',
    templateUrl: 'portal/welcome/steps/credentials.html',
    bindings: {
      token: '<',
      club: '<',
      user: '<',
    },
  })
```

I don't want the config and component blocks to be indented. 3.x didn't use to do this, and I can't seem to find a setting in the indent rule to turn this behaviour off. Please advise.

I've had to revert to 3.x to prevent ESLint from messing up all my files.

**Comment by platinumazure** [comment]
Hi @adamreisnz, thanks for the issue.

Have you tried the MemberExpression setting?

```yaml
  indent:
    - error
    - 2
    - SwitchCase: 1
      VariableDeclarator:
        let: 2
        const: 3
      MemberExpression: 0  // or off, which is closer to ESLint 3.x
```

In future, when upgrading major versions, you would do well to read our Migration Guide. [Here](https://eslint.org/docs/user-guide/migrating-to-4.0.0) is the 4.0.0 migration guide. Hope this helps!

**Comment by adamreisnz** [comment]
Hi @platinumazure , thanks for this. I have extensively read the documentation, but not the migration guide.

Could you please add a description of the `MemberExpression` to the online documentation?

**Comment by not-an-aardvark** [comment]
There is a description of the `MemberExpression` option in the [rule documentation](http://eslint.org/docs/rules/indent#options).

> I've had to revert to 3.x to prevent ESLint from messing up all my files.

If you just want to replicate the `indent` behavior from 3.x, you can use the `indent-legacy` rule, which is a copy of the `indent` rule from 3.x. However, the updated `indent` rule checks indentation in many places where `indent-legacy` does not, so it might be desirable if you want to check indentation more strictly.

**Comment by adamreisnz** [comment]
In addition, I'm afraid setting this to 0 doesn't help. Now ESLint breaks the indentation where I *do* want members to be indented:

Turns code like:

```js
//Save
return this.club
  .updateWelcomeMessage(welcomeMessage)
  .then(() => $notice.show('Welcome message updated'));
```

Into:

```js
//Save
return this.club
.updateWelcomeMessage(welcomeMessage)
.then(() => $notice.show('Welcome message updated'));
```

Is there no way to tweak the settings to have the behaviour how it was in ESLint 3? Basically just have it not touch my indentation of members at all?

**Comment by not-an-aardvark** [comment]
You can use `MemberExpression: "off"`, or use the `indent-legacy` rule.

**Comment by adamreisnz** [comment]
> Configuration for rule "indent" is invalid: Value "off" is the wrong type.

**Comment by not-an-aardvark** [comment]
Are you sure you're using ESLint 4.x? You mentioned you downgraded back to 3.x, and the `"off"` option was only introduced in 4.0.

**Comment by adamreisnz** [comment]
Yeah I installed 4 again, but failed to restart Atom. It started erroring on the indentation again so I assumed it was using 4. After restart the 'off' setting works and it's not touching the indentation, thanks.
